wapiti/parsers.py

```diff
@@ -154,8 +154,12 @@ def __init__(self, format, jsonp=None, serialize_all_fields=False,
         self.serialize_all_fields = serialize_all_fields
         self.max_depth = max_depth
         self.file_handler = file_handler
-        filehandler_args = getargspec(self.file_handler)[0]
-        self._filehandler_backwards_compat = len(filehandler_args) > 1
+        try:
+            filehandler_args = getargspec(self.file_handler)[0]
+            self._filehandler_backwards_compat = len(filehandler_args) > 1
+        except TypeError:
+            # Looks like self.file_handler isn't a function
```

Comment:
I'm not a fan of assuming things through an exception, but if `self.file_handler` is something weird, when it gets called, the right exception will pop up, so this should be fine.